packages/flutter_driver/test/src/real_tests/timeline_summary_test.dart

```diff
@@ -734,5 +750,181 @@ void main() {
+                                                                  startTime: startTimeInNanoseconds,
+                                                                  interval: intervalInNanoseconds,
+                                                                  margin: 0,
+                                                                 );
+        final RefreshRateSummarizer summarizer = _summarize(events);
+        final RefreshRateSummary summary = summarizer.computeRefreshRateSummary();
+        expect(summary.percentageOf30HzFrames, 0);
+        expect(summary.percentageOf60HzFrames, 0);
+        expect(summary.percentageOf90HzFrames, 0);
+        expect(summary.percentageOf120HzFrames, 0);
+        expect(summary.framesWithIllegalRefreshRate, isNotEmpty);
+        expect(summary.framesWithIllegalRefreshRate.first, closeTo(100, _kCompareDelta));
+      });
+
+      test('Mixed refresh rates.', () async {
+
+        final List<Map<String, dynamic>> events = <Map<String, dynamic>>[];
+        const int num30Hz = 10;
+        const int num60Hz = 20;
+        const int num90Hz = 20;
+        const int num120Hz = 40;
+        const int numIllegal = 10;
+
+        // Add 30hz frames
+        events.addAll(_populateEvents(numberOfEvents: num30Hz,
+                                      startTime: 0,
+                                      interval: 32000000,
+                                      margin: 0,
+                                      ));
+
+        // Add 60hz frames
+        events.addAll(_populateEvents(numberOfEvents: num60Hz,
+                                      startTime: 0,
+                                      interval: 16000000,
+                                      margin: 0,
+                                      ));
+
+
+        // Add 90hz frames
+        events.addAll(_populateEvents(numberOfEvents: num90Hz,
+                                      startTime: 0,
+                                      interval: 11000000,
+                                      margin: 0,
+                                      ));
+
+        // Add 120hz frames
+        events.addAll(_populateEvents(numberOfEvents: num120Hz,
+                                      startTime: 0,
+                                      interval: 8000000,
+                                      margin: 0,
+                                      ));
+
+        // Add illegal refresh rate frames
+        events.addAll(_populateEvents(numberOfEvents: numIllegal,
+                                      startTime: 0,
+                                      interval: 60000,
+                                      margin: 0,
+                                      ));
+
+        final RefreshRateSummarizer summarizer = _summarize(events);
+        final RefreshRateSummary summary = summarizer.computeRefreshRateSummary();
+        expect(summary.percentageOf30HzFrames, closeTo(10, _kCompareDelta));
```

Comment:
use numNNHz values here instead of hard coded numbers?

Reply:
done!